ETrade/ETrade/EShop.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ETrade
+{
+    public class EShop
+    {
+        List<Order> orders;
+        OrderCreator creator;
+        int nextID;
+
+        public EShop(OrderCreator creator)
+        {
+            this.orders = new List<Order>();
+            this.creator = creator;
+            nextID = 1;
+        }
+
+        public void AddOrder(string record)
+        {
+            orders.Add(creator.Create(nextID++, record));
+        }
+
+        public void RemoveOrder(int id)
+        {
+            var order = orders.FirstOrDefault(o => o.ID == id);
+            orders.Remove(order);
+        }
+
+
+        public void ExecuteOrders(string category, string name, int totalQuantity)
+        {
+            orders = orders
+                .Where(order => order.Goods.Category != category || order.Goods.Name != name)
+                .Concat(
+                    orders
+                        .Where(order => order.Goods.Category == category && order.Goods.Name == name)
+                        .OrderBy(order => order.ID)
+                        .Select(order =>
+                        {
+                            if (totalQuantity > 0)
+                            {
+                                if (order.Quantity <= totalQuantity)
+                                {
+                                    totalQuantity -= order.Quantity;
+                                    return null;
+                                }
+                                else
+                                {
+                                    order.Quantity -= totalQuantity;
+                                    totalQuantity = 0;
+                                }
+                            }
+                            return order;
+                        })
+                        .Where(order => order != null)
+                )
+                .ToList();
+        }
+        public int Count(string category, string name)
+        {
+            return orders.Where(order => order.Goods.Category == category).Sum(order => order.Quantity);
```

Comment:
Цепочка вызовов более читаема в несколько строк
```suggestion
            return orders
                .Where(order => order.Goods.Category == category)
                .Sum(order => order.Quantity);
```